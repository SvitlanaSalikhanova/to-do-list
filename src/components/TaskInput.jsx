import React from 'react';
import TaskInputCss from '../styles/TaskInput.module.scss';

export default function TaskInput({ tempText, changeText, addItem }) {
    return (
        <input
            type="text"
            placeholder="Enter your task and press ENTER"
            className={TaskInputCss.inputTask}
            value={tempText}
            onChange={changeText}
            onKeyUp={addItem}
        />
    );
}
