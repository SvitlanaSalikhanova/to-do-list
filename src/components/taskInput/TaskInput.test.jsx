import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import TaskInput from './TaskInput';

const tempText = 'text';

const onChangeMock = jest.fn();
const onAddItemMock = jest.fn();

describe('TaskInput component tests', () => {
    it('TaskInput Matches Snapshot', () => {
        const domTree = renderer.create(
            <TaskInput tempText={tempText} changeText={onChangeMock} addItem={onAddItemMock} />,
        ).toJSON();
        expect(domTree).toMatchSnapshot();
    });

    it('when changes calls onChange', () => {
        render(
            <TaskInput tempText={tempText} changeText={onChangeMock} addItem={onAddItemMock} />,
        );
        const input = screen.getByRole('textbox');
        fireEvent.change(input, { target: { value: 'test' } });
        expect(onChangeMock).toBeCalledTimes(1);
    });

    it('when changes calls onChange', () => {
        render(
            <TaskInput tempText={tempText} changeText={onChangeMock} addItem={onAddItemMock} />,
        );
        const input = screen.getByRole('textbox');
        fireEvent.keyUp(input);
        expect(onAddItemMock).toBeCalledTimes(1);
    });

    // it('on keyup calls onAddItem', () => {
    //     render(
    //         <Provider store={store}>
    //             <TaskInput tempText={tempText} onChange={onChangeMock} onKeyUp={onAddItemMock} />
    //         </Provider>,
    //     );
    //     const input = screen.getByRole('textbox');
    //     fireEvent.keyUp(input);
    //     expect(onAddItemMock).toBeCalledTimes(1);
    // });
});
