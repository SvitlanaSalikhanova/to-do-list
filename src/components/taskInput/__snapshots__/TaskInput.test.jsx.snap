// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskInput component tests TaskInput Matches Snapshot 1`] = `
<input
  className="inputTask"
  onChange={[MockFunction]}
  onKeyUp={[MockFunction]}
  placeholder="Enter your task and press ENTER"
  type="text"
  value="text"
/>
`;
